auth/controllers/userController.ts

```diff
@@ -0,0 +1,79 @@
+import { Request, Response } from "express";
+import { getRepository } from "typeorm";
+import User from "../models/user";
+import { addUserToDB, authUser } from "../services/AuthService";
+import { hashPassword, ReE, ReS, to } from "../services/Util";
+
+export async function loginUser(req: Request, res: Response): Promise<void> {
+  res.setHeader("Content-Type", "application/json");
+
+  const body = req.body;
+  let err;
+  let user: any;
+
+  [err, user] = await to(authUser(body));
+  if (err) {
+    return ReE(res, err, 422);
+  }
+  return ReS(res, { token: user.getJWT(), user: user.toJSON() }, 200);
+}
+
+export async function createUser(req: Request, res: Response): Promise<void> {
+  res.setHeader("Content-Type", "application/json");
+
+  const body = req.body;
+  let err;
+  let user: any;
+
+  if (!body.password) {  return ReE(res, "No password was entered", 422); }
+  // Hash password before saving to db
+  body.password = await hashPassword(body.password);
```

Comment:
This line is causing the program to crash for me when creating an account. Does this not happen for you? The password is a valid string. It actually crashes on the bcrypt.hash(password, saltRounds) function in Util.ts.

Reply:
Hmmm, that's weird it works fine on my system .. are you sure bcrypt is installing successfully? 

Can you provide me with 
- Exact Input conditions that caused the error 
- Error Log
- npm version 
- npm install log 
- npm run build log
- System details 

I would try to recreate the exact conditions and test .. or if you want we can check it out on sat

Reply:
![bcrypt](https://user-images.githubusercontent.com/31572141/52821161-d5003680-3062-11e9-81f0-e516088987b1.PNG)

Reply:
I had the same problem, and saw this issue on Github: https://github.com/kelektiv/node.bcrypt.js/issues/16 
`npm rebuild bcrypt --build-from-source` worked for me. It looks like there's an issue in the version of bcrypt we're using?

<img width="972" alt="screen shot 2019-02-15 at 1 09 42 pm" src="https://user-images.githubusercontent.com/12488887/52884503-07716880-3123-11e9-945a-d571a2d876ac.png">

Reply:
@frostyshadows @cowmanjoe 

I have tried to recreate the error on 
- ubuntu linux vm (auto-build didn't fail) 👍 
- windows 10 edu vm (auto-build failed) 👎 

I think you guys are right ! I checked it out and it turns the latest version needs to be built from source and most of the time the auto-build fails .. 

I have tried some other versions of the module and @2.0.1 doesn't fail the auto build. I will look into more versions and investigate it further and reach back to you guys on Saturday.

Anyway, when I hook everything up on docker I will fix the auto-build issue with a pre-install script which would be customized to suite the docker's vm os and ecosystem

Reply:
@cowmanjoe 
Try bcrypt@2.0.1
 or 
```bash
foo@bar:~$npm rebuild bcrypt --build-from-source
```
or 
Try testing with docker-compose (I have just added support for that)